apstools/beamtime/manage_ioc.sh

```diff
@@ -0,0 +1,174 @@
+            # compare directories
+            BIN_CWD=$(readlink "/proc/${pid}/cwd")
+            IOC_CWD=$(readlink -f "${IOC_STARTUP_DIR}")
+
+            if [ "$BIN_CWD" = "$IOC_CWD" ] ; then
+                # The IOC is running;
+                # the process with PID=$pid is the
+                # IOC that was run from $IOC_STARTUP_DIR
+                P_PID=$(ps -p "${pid}" -o ppid=)
+                # strip leading (and trailing) whitespace
+                arr=($P_PID)
+                P_PID=${arr[0]}
+                SCREEN_SESSION="${P_PID}.${SESSION_NAME}"
+                SCREEN_MATCH=$(screen -ls "${SCREEN_SESSION}" | grep "${SESSION_NAME}")
+                if [ "${SCREEN_MATCH}" != "" ] ; then
+                    # IOC is running in screen
+                    IOC_DOWN=0
+                    IOC_PID=${pid}
+                    SCREEN_PID=${P_PID}
+                    break
+                fi
+            fi
+        done
+    else
+        # IOC is not running
+        IOC_DOWN=1
+    fi
+
+    return ${IOC_DOWN}
+}
+
+function console () {
+    if checkpid; then
+        echo "Connecting to ${SCREEN_SESSION}'s screen session"
+        # The -r flag will only connect if no one is attached to the session
+        #!screen -r "${SESSION_NAME}"
+        # The -x flag will connect even if someone is attached to the session
+        screen -x "${SCREEN_SESSION}"
+    else
+        echo "${SCREEN_NAME} is not running"
+    fi
+}
+
+function exit_if_running() {
+    # ensure that multiple, simultaneous IOCs are not started by this user ID
+    MY_UID=$(id -u)
+    IOC_PID=$(pgrep "${SESSION_NAME}"\$ -u "${MY_UID}")
+
+    if [ "" != "${IOC_PID}" ] ; then
+        echo "${SESSION_NAME} IOC is already running (PID=${IOC_PID}), won't start a new one"
+        exit 1
+    fi
+}
+
+function restart() {
+    stop
+    start
+}
+
+function run_ioc() {
+    exit_if_running
+    ${START_IOC_COMMAND}
+}
+
+function screenpid() {
+    if [ -z "${SCREEN_PID}" ] ; then
+        echo
+    else
+        echo " in a screen session (pid=${SCREEN_PID})"
+    fi
+}
+
+function start() {
+    if checkpid; then
+        echo -n "${SCREEN_SESSION} is already running (pid=${IOC_PID})"
+        screenpid
+    else
+        echo "Starting ${SESSION_NAME}"
+        cd "${IOC_STARTUP_DIR}"
+        # Run SESSION_NAME inside a screen session
+        screen -dm -S "${SESSION_NAME}" -h 5000 "${START_IOC_COMMAND}"
+    fi
+}
+
+function status() {
+    if checkpid; then
+        echo -n "${SCREEN_SESSION} is running (pid=${IOC_PID})"
+        screenpid
+    else
+        echo "${SESSION_NAME} is not running"
+    fi
+}
+
+function stop() {
+    if checkpid; then
+        echo "Stopping ${SCREEN_SESSION} (pid=${IOC_PID})"
+        kill "${IOC_PID}"
+    else
+        echo "${SESSION_NAME} is not running"
+    fi
+}
+
+function usage() {
+    echo "Usage: $(basename ${SHELL_SCRIPT_NAME}) {start|stop|restart|status|console|run} [NAME [PREFIX]]"
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/gh/BCDA-APS/apstools/pullRequest?prid=5764884)